src/codegen.rs

```diff
@@ -270,10 +269,10 @@ impl<'ink> GeneratedModule<'ink> {
             FormatOption::Object | FormatOption::Relocatable => {
                 self.persist_as_static_obj(output, target, optimization_level)
             }
-            FormatOption::PIC | FormatOption::Static => {
+            FormatOption::PIC | FormatOption::Shared | FormatOption::Static => {
                 self.persist_to_shared_pic_object(output, target, optimization_level)
             }
-            FormatOption::Shared => self.persist_to_shared_object(output, target, optimization_level),
+            FormatOption::NoPIC => self.persist_to_shared_object(output, target, optimization_level),
```

Comment:
Not sure if we want to tackle this in this PR (or at all), but we can have some code-reduction here by passing a `RelocMode` variant directly to the `persist_to_obj` instead of calling the "wrapper" functions `persist_as_static_obj`, `persist_to_shared_pic_object` and `persist_to_shared_object`.

Reply:
Yes, let's discuss that option. We would still have to wrap a RelocMode since it's inkwell API which i'm trying not to expose outside the codegen modules

Reply:
Might be missing something, but this is all codegen related code anyways so `RelocMode` would never be exposed? I was thinking of refactoring the match statement to 
```
FormatOption::Object | FormatOption::Relocatable => {
    self.persist_to_obj(output, RelocMode::Default, target, optimization_level)
}
FormatOption::PIC | FormatOption::Static => {
    self.persist_to_obj(output, RelocMode::PIC, target, optimization_level)
}
FormatOption::Shared => {
    self.persist_to_obj(output, RelocMode::DynamicNoPic, target, optimization_level)
}
```
removing the then unused wrapper functions. But that's a super minor nitpick anyways :P